src/jit/emitarm.cpp

```diff
@@ -4928,31 +4917,44 @@ inline unsigned insEncodeImmT2_Mov(int imm)
     return result;
 }
 
-/*****************************************************************************
- *
- *  Unscales the immediate based on the operand size in 'size'
- */
-/*static*/ int emitter::insUnscaleImm(int imm, emitAttr size)
+//------------------------------------------------------------------------
+// insUnscaleImm: Unscales the immediate operand of a given instruction
+//
+// Arguments:
+//    ins - the instruction
+//    fmt - the instruction format
+//    imm - the immediate value to unscale
+//
+// Return Value:
+//    The unscaled immediate value
+//
+// Notes:
+//    Only IF_T1_C encoded instructions are supported.
+//
+/*static*/ int emitter::insUnscaleImm(instruction ins, insFormat fmt, int imm)
```

Comment:
Seems a bit odd the have a parameter that is always required to be IF_T1_C

Do you believe that we will extend support to include more that one instruction format here?

Reply:
Right, it's unlikely to add more instructions formats here.

AFAIR this is just a side effect of the way I refactored the code - first I added `ins` + `fmt` and removed `size` parameter, at that point it was still supporting more formats. Then I realized that all formats but `IF_T1_C` had trivial scaling needs, similar to that of many other formats that did some kind of scaling without using `insUnscaleImm`. So I removed everything but `IF_T1_C` and added an assert to be sure the function doesn't get misused by accident.

I suppose I could go one step further - rename it to `insUnscaleImm_IF_T1_C` and remove the format parameter?

Reply:
I don't think that it needs to be renamed, the extra parameter can just be removed.
We still have this assert which prevents unexpected callers from using this method:

```
       default:
            assert(!"Invalid value in size");	            assert(!"Invalid IF_T1_C instruction");
```

Reply:
Right, parameter removed.